src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -1836,6 +1860,14 @@ class ArnoldRenderer : public IECoreScenePreview::Renderer
 				try
 				{
 					m_outputs[name] = new ArnoldOutput( name, output, nodeDeleter( m_renderType ) );
+					if( m_outputs[name]->lpeValue().size() )
```

Comment:
Hmm, yeah, I should only be generating the explicit list of lpes into temp data - that totally makes sense, especially now that the ArnoldOutput has a convenient way to grab the lpe, which wasn't the case when I first wrote it.  Should be an easy clean up.

Reply:
While you're at it, do we need to return `lpeName()` and `lpeValue()` separately and then do the formatting outside of ArnoldOutput? The previous design had everything encapsulated so that `ArnoldOutput` was the only thing that really knew the details of outputs and their formatting, and then ArnoldRenderer just had to build the final parameter arrays. Maybe we should replace the `ArnoldOutput::string()` and `ArnoldOutput::lpe*()` methods with a single `ArnoldOutput::append( vector<string> &outputs, vector<string> &light_path_expressions )` method? It's a minor thing, but I think it's a worthwhile improvement given that we're rejigging this a bit anyway...